packages/miniflare/test/index.spec.ts

```diff
@@ -96,22 +96,40 @@ test("Miniflare: ready returns copy of entry URL", async (t) => {
   t.is(url2.protocol, "http:");
 });
 
-test("Miniflare: keeps port between updates", async (t) => {
+test("Miniflare: setOptions: can update host/port", async (t) => {
+  // Extract loopback port from injected live reload script
+  const loopbackPortRegexp = /\/\/ Miniflare Live Reload.+url\.port = (\d+)/s;
+
   const opts: MiniflareOptions = {
     port: 0,
+    liveReload: true,
     script: `addEventListener("fetch", (event) => {
-      event.respondWith(new Response("a"));
+      event.respondWith(new Response("<p>👋</p>", {
+        headers: { "Content-Type": "text/html;charset=utf-8" }
+      }));
     })`,
   };
   const mf = new Miniflare(opts);
   t.teardown(() => mf.dispose());
+
   const initialURL = await mf.ready;
+  let res = await mf.dispatchFetch("http://localhost");
+  const initialLoopbackPort = loopbackPortRegexp.exec(await res.text())?.[1];
 
+  opts.host = "0.0.0.0";
   await mf.setOptions(opts);
   const updatedURL = await mf.ready;
+  res = await mf.dispatchFetch("http://localhost");
+  const updatedLoopbackPort = loopbackPortRegexp.exec(await res.text())?.[1];
 
+  // Make sure a new was allocated when `port: 0` was passed to `setOptions()`
   t.not(initialURL.port, "0");
-  t.is(initialURL.port, updatedURL.port);
+  t.not(initialURL.port, updatedURL.port);
```

Comment:
I don't think this is actually what we were shooting for.

I think the previous behaviour of: if option `port: 0` was set originally and remained `0` on `setOptions`, retaining the original port is very helpful and expected.

What we need for wrangler, if option `port` was set to something other that `0` originally and changed to another number (anything, even `0`), the port must change (the bug before was that miniflare retained the previous port, even if both times they were non zero and different)

Reply:
I agree that race condition is quite unlikely. More so because the port was recently bound by ourselves so we can assume we can rebind to it

The DX of changing the existing config to reset the config without changes is suboptimal. The user would also have to do the same for the inspectorPort and the unsafeDirectPorts on each subworker which gets even harder.

I made this ticket https://jira.cfdata.org/browse/DEVX-927